### PR TITLE
Make E2E validation more flexible.

### PR DESCRIPTION
**Pull request by r2k1**
Previously, E2E scenarios were limited to executing shell scripts on the VM.

I updated the tests to support more complex validations, such as running a GPU-enabled pod and verifying its functionality.

This change made the tests more flexible and simplified the overall logic.

### CODE REVIEW

File: e2e/types.go [hunk 1]
```diff
@@ -142,9 +142,7 @@ type Config struct {
 	// VMConfigMutator is a function which mutates the base VMSS model according to the scenario's requirements
 	VMConfigMutator func(*armcompute.VirtualMachineScaleSet)
 
-	// LiveVMValidators is a slice of LiveVMValidator objects for performing any live VM validation
-	// specific to the scenario that isn't covered in the set of common validators run with all scenarios
-	LiveVMValidators []*LiveVMValidator
+	Validator func(ctx context.Context, s *Scenario)
```
Comment (r2k1) [inline e2e/types.go:145]: This is the core difference

File: e2e/scenario_helpers_test.go [hunk 2]
```diff
@@ -121,42 +121,46 @@ func maybeSkipScenario(ctx context.Context, t *testing.T, s *Scenario) {
 			t.Fatalf("could not find image for %q: %s", t.Name(), err)
 		}
 	}
-	t.Logf("running scenario %q with vhd: %q, tags %+v", t.Name(), vhd, s.Tags)
+	t.Logf("running scenario vhd: %q, tags %+v", vhd, s.Tags)
```
Comment (djsly) [inline e2e/scenario_helpers_test.go:124]: the scenario name wasn't useful ?

File: e2e/types.go (hunk 1 above)
Comment (djsly) [reply e2e/types.go:149]: could you explain the main reason for this change ?

File: e2e/scenario_helpers_test.go (hunk 2 above)
Comment (r2k1) [reply e2e/scenario_helpers_test.go:124]: each log message is associated with a test. It's just a duplication of the available information.

**Review by djsly — APPROVED** [review]

**Pull request merged**